src/windows.c

```diff
@@ -506,10 +506,18 @@ static void draw_window_tab(ToxWindow *toxwin, bool active_window)
 
     clrtoeol();
 
-    if (active_window || toxwin->index <= 1) {
+    WINDOW_TYPE type = toxwin->type;
+
+    if (active_window || (type == WINDOW_TYPE_PROMPT || type == WINDOW_TYPE_FRIEND_LIST)) {
         printw(" [%s]", toxwin->name);
     } else {
-        printw(" [%u]", toxwin->index - 1);
+        unsigned int pending_messages = toxwin->pending_messages;
```

Comment:
You write to toxwin->pending_messages after locking the winthread lock. Does this read need to be guarded too?